src/views/Detail.vue

```diff
@@ -75,52 +76,63 @@
           v-for="(item, index) in items"
           :key="index"
         >
-          <ImageDetail
-            v-if="fileType === 'images'"
-            :file="item"
-          />
-          <DirectoryDetail
-            v-if="fileType === 'directories'"
-            :file="item"
-          />
-          <DocumentDetail
-            v-if="fileType === 'text'"
-            :file="item"
-          />
-          <AudioDetail
-            v-if="fileType === 'audio'"
-            :file="item"
-          />
-          <VideoDetail
-            v-if="fileType === 'video'"
+          <component
+            :is="componentType"
             :file="item"
           />
         </v-carousel-item>
       </v-carousel>
+      <component
+        v-else
+        :is="componentType"
+        :file="singleItem"
+      />
     </div>
   </div>
 </template>
 
 <script>
-import AudioDetail from '@/components/results/detail/AudioDetail';
-import DocumentDetail from '@/components/results/detail/DocumentDetail';
-import VideoDetail from '@/components/results/detail/VideoDetail';
-import DirectoryDetail from '@/components/results/detail/DirectoryDetail';
-import ImageDetail from '@/components/results/detail/ImageDetail';
```

Comment:
These @-imports seem a lot more readable and consistent with style elsewhere.

Reply:
I agree. I think there has been some weird automatic import renamer, either eslint or webstorm or the two conspiring together somehow.